web/packages/teleport/src/Welcome/Welcome.tsx

```diff
@@ -24,6 +24,13 @@ import cfg from 'teleport/config';
 import { NewCredentials } from './NewCredentials';
 import { CardWelcome } from './CardWelcome';
 
+/**
+ *
+ * @remarks
+ * This component is duplicated in Enterprise for Enterprise onboarding. If you are making edits to this file, check to see if the
+ * equivalent change should be applied in Enterprise
+ *
+ */
```

Comment:
Did you consider making `Welcome` accept `NewCredentials` as a prop? teleport.e could then import `Welcome` from OSS and pass a different `NewCredentials` implementation.

Reply:
I think the same might be possible for `NewCredentials` itself. Instead of copying the implementation wholesale to add that one different bit, `NewCredentials` could accept a component through a prop, called say `$onSuccess`¹.

OSS could pass `RegisterSuccess`  there while E would pass a special wrapper that does `!resetMode && displayOnboardingQuestionnaire` and displays either `Questionnaire` or `RegisterSuccess`.

This way we'd avoid duplication completely, unless I'm missing some details.

¹ Here's [a discussion about this naming scheme with $](https://gravitational.slack.com/archives/C02DQ1C2BMW/p1682344664575919).